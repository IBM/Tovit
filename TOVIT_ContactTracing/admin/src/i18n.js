import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import en from './translations/en'

// the translations
// (tip move them in a JSON file and import them)
const resources = {
  en
}

i18n
  .use(LanguageDetector)
  .use(initReactI18next) // passes i18n down to react-i18next
  .init({
    resources,
    fallbackLng: 'en',
    keySeparator: false, // we do not use keys in form messages.welcome
    interpolation: {
      escapeValue: false // react already safes from xss
    },
    detection: {
      order: ['querystring', 'cookie', 'localStorage', 'navigator', 'htmlTag', 'path', 'subdomain'],

      // keys or params to lookup language from
      lookupQuerystring: 'lng',
      lookupCookie: 'i18next',
      lookupLocalStorage: 'i18nextLng',
      lookupFromPathIndex: 0,
      lookupFromSubdomainIndex: 0,

      // cache user language on
      caches: ['localStorage', 'cookie'],
      excludeCacheFor: ['cimode'], // languages to not persist (cookie, localStorage)

      // optional expire and domain for set cookie
      cookieMinutes: 10,
      cookieDomain: 'myDomain',

      // optional htmlTag with lang attribute, the default is:
      htmlTag: document.documentElement,

      // only detect languages that are in the whitelist
      checkWhitelist: true
    }
  })

export default i18n
