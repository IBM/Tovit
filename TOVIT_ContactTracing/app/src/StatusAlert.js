import React from 'react'
import Alert from '@material-ui/lab/Alert'
import { Translation } from 'react-i18next'

class StatusAlert extends React.Component {
  render () {
    const { status, onExposuresTab } = this.props
    const extraText = onExposuresTab ? '' : (<Translation>{t => ' ' + t('seeStatusTabMessage')}</Translation>)
    if (status) {
      return (
        <Alert style={{ marginTop: 25, width: '100%' }} severity='error'><Translation>{t => t('elevatedRiskAlertMessage')}</Translation>{extraText}</Alert>
      )
    } else {
      return <div />
    }
  }
}

export default StatusAlert
