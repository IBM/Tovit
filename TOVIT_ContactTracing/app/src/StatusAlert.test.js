/* globals test, expect */
import React from 'react'
import StatusAlert from '../src/StatusAlert'
import TestRenderer from 'react-test-renderer'
import { I18nextProvider } from 'react-i18next'
import i18n from '../src/i18n'

test('StatusAlert component: on exposures tab, not exposed', () => {
  const status = false
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <StatusAlert status={status} onExposuresTab />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})

test('StatusAlert component: on exposures tab, exposed', () => {
  const status = true
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <StatusAlert status={status} onExposuresTab />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})

test('StatusAlert component: not on exposures tab, not exposed', () => {
  const status = false
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <StatusAlert status={status} />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})

test('StatusAlert component: not on exposures tab, exposed', () => {
  const status = true
  const component = TestRenderer.create(
    <I18nextProvider i18n={i18n}>
      <StatusAlert status={status} />
    </I18nextProvider>)
  expect(component).toMatchSnapshot()
})
