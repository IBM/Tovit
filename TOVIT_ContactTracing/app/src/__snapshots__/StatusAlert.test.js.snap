// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusAlert component: not on exposures tab, exposed 1`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
  role="alert"
  style={
    Object {
      "marginTop": 25,
      "width": "100%",
    }
  }
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  >
    Your risk level is elevated.
     See the status tab.
  </div>
</div>
`;

exports[`StatusAlert component: not on exposures tab, not exposed 1`] = `<div />`;

exports[`StatusAlert component: on exposures tab, exposed 1`] = `
<div
  className="MuiPaper-root MuiAlert-root MuiAlert-standardError MuiPaper-elevation0"
  role="alert"
  style={
    Object {
      "marginTop": 25,
      "width": "100%",
    }
  }
>
  <div
    className="MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message"
  >
    Your risk level is elevated.
    
  </div>
</div>
`;

exports[`StatusAlert component: on exposures tab, not exposed 1`] = `<div />`;
